### PR TITLE
[mysql] Improve function documentation

### PR DESCRIPTION
**Pull request by alexjurkiewicz**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.) **No test changes**
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
```
ERROR: 216:8  unified-signatures  This overload and the one on line 208 can be combined into one signature taking `'connection' | 'acquire'`.
ERROR: 224:8  unified-signatures  This overload and the one on line 208 can be combined into one signature taking `'connection' | 'release'`.
ERROR: 224:8  unified-signatures  This overload and the one on line 216 can be combined into one signature taking `'acquire' | 'release'`.
```
**These have been split up to allow different function docs.** Is it OK to add ignore declaration for these?

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/mysqljs/mysql
- [x] Increase the version number in the header if appropriate. **No type definition changes.**
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. **Is already done**

### CODE REVIEW

**Comment by alexjurkiewicz** [comment]
I added a commit to ignore the lint errors. Still not sure if it's the right approach though.

File: types/mysql/index.d.ts [hunk 1]
```diff
@@ -535,37 +667,97 @@ export interface MysqlError extends Error {
 }
 
 export const enum Types {
```
Comment (kpping) [inline types/mysql/index.d.ts]: On this `Enum: Types`, I think we should leave it as is until we can precisely document them. What do you think ?

**Review by kpping — COMMENTED** [review]
- I think the return type (in the comments) is incorrect.
- For `Enum: Types` part, I think leave it as is might be better. What do you think ?
- I think `tslint:disable-next-line:unified-signatures` is acceptable in this case.

Thank you for your contribution.

File: types/mysql/index.d.ts (hunk 1 above)
Comment (alexjurkiewicz) [reply types/mysql/index.d.ts]: that's fine with me

**Comment by alexjurkiewicz** [comment]
Thank you for the review :tada: . I removed the Enum type conversion as you suggested, and fixed the three errors.

**Review by kpping — APPROVED** [review]
Thank you for your contribution. 🎉

**Comment by PranavSenthilnathan** [comment]
Congratulations on your first DefinitelyTyped contribution! Thank you for being a part of the community!

**Pull request merged**